mne/io/neuralynx/tests/test_neuralynx.py

```diff
@@ -220,3 +220,17 @@ def test_neuralynx_gaps():
     assert_allclose(
         mne_y, mat_y, rtol=1e-6, err_msg="MNE and Nlx2MatCSC.m not all close"
     )
+
+    # test that channel selection works
+    raw = read_raw_neuralynx(
+        fname=testing_path,
+        preload=False,
+        exclude_fname_patterns=ignored_ncs_files,
+    )
+
+    raw.pick("LAHC2")
+    assert raw.ch_names == ["LAHC2"]
+    try:
+        raw.load_data()
+    except Exception:
+        pytest.fail("Loading data fails after picking one channel with gaps.")
```

Comment:
If we try-excepted every smoke test like this, our test suite would be too big and too hard to read. So let's just let pytest raise the exception if there is one
```suggestion
    raw.load_data()  # before gh-12357 this would fail
```